gobblin-cluster/src/main/java/org/apache/gobblin/cluster/SleepingTask.java

```diff
@@ -73,10 +76,10 @@ public void run() {
     } catch (InterruptedException e) {
       log.error("Sleep interrupted.");
       Thread.currentThread().interrupt();
-      Throwables.propagate(e);
+      failTask(e, taskContext);
```

Comment:
SleepingTask was written to test cancellation code or hanging issues. It is meant to sleep for some amount to simulate as a hanging task in production. It is supposed propagate exceptions like any normal task would do.